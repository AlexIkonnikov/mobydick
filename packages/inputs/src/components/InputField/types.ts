import {ITextInputProps} from '@npm/mobydick-core';
import {ReactElement} from 'react';
import {IStyledTextProps} from '@npm/mobydick-typography';
import {StyleProp, ViewStyle} from 'react-native';

import {ITypes} from '../types';

export interface IInputFieldsProps extends ITextInputProps {
  title?: string;
  titleProps?: IStyledTextProps;
  subtitle?: string;
  subtitleProps?: IStyledTextProps;
  containerStyle?: StyleProp<ViewStyle>;
  textInputContainerStyle?: StyleProp<ViewStyle>;
  rightIcon?: ReactElement;
  type?: ITypes;
  disabled?: boolean;
}
