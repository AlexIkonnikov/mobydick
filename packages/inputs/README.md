## @npm/mobydick-inputs

Author - Aleksei Smirnov
# ****DropDown правила кастомизации****

1. Процентные стили не поддерживаются!
2. Необходимо поставить [React-Native-Device-Info](https://github.com/react-native-device-info/react-native-device-info)
3. Горизонтальная верстка пока не поддерживается, но можно кастомизировать пропсами
4. Лучше выбирать либо **TextStyle** либо **TextFont**
5. Для **FlatList** ширина берется от кнопки стандартно, можно передать свою
6. Для **FlatList** высота **Item** высчитывается по формуле: **Высота кнопки * 0.8 (как в дизайне)**, можно передать свою
7. Что за функция такая **getIosSafeAreaHeights** и зачем она такая нужна:
   * У всех устройств есть **Status Bar**, у iPhone начиная с версии iPhone X есть еще **Bottom Line**,
   Для определения кнопки в пространстве используется ее **ref** и функция для получения позиции на экране **ref.current.measure**.
   iPhone возвращает позицию **pageY** учитывая высоты **Status Bar** и **Bottom line** в отличие от Android, поэтому приходится принудительно в функции **getDropDownDimensions** 
   добавлять значения отступов. Мы получаем модель телефона через **getModel** (react-native-device-info) и пробрасываем его в функцию **getIosSafeAreaHeights**.
   Для этой функции создан enum **IPhoneModelEnum** и функция после перебора возвращает для определенных моделей iPhone заранее записанные значения отступов.

      `!!В дальнейшем, для поддержания актуальности библиотеки, нужно будет
      актуализировать enum IPhoneModelEnum и функцию getIosSafeAreaHeights!!`

8. Расшифровка пропсов: 
   * label?: string; - Тест в заголовке над **DropDown**
   * rightIcon?: ReactElement; - Иконка для **DropDown**
   * placeholder: string; - Дефолтный текст в **DropDown**
   * list: string[]; - Массив элементов для списка **DropDown**
   * onPress: (item: string) => void; - Функция по нажатию на элемент массива
   * selectedItem?: string; - Текущий выбранный элемент для окраски оного в DropDown
   * navBarHeight?: number; - Высота навигационного бара для правильной позиции, чтобы **DropDown** не заезжал под навигацию
   * addButtonStyle?: ViewStyle; - Кастомный стиль для кнопки - если передать только его, **item** в списке будут считать высоту по **"высота кнопки * 0.8"**
   * addFlatListStyle?: ViewStyle; - Кастомный стиль для вьюшки **FlatList**. Стандартно ширина берется от кнопки, высота высчитывается в функции
   * addFlatListItemStyle?: ViewStyle; - Кастомный стиль для RenderItem у **FlatList**
   * addLabelStyle?: TextStyle; - кастомный стиль для **label**, рекомендуется не использовать отступы
   * addLabelFont?: TypographyProp; - кастомный шрифт из дизайн системы для **label**
   * addButtonTextStyle?: TextStyle; - кастомный стиль текста для кнопки с дефолтным **Placeholder**
   * addButtonTextStyleChosen?: TextStyle; - кастомный стиль текста для кнопки с выбранным элементом - **selectedItem**
   * addButtonTextFont?: TypographyProp; - кастомный шрифт для текста кнопки с дефолтным **Placeholder**
   * addButtonTextFontChosen?: TypographyProp; - кастомный шрифт для текста кнопки при выбранном элементе - **selectedItem**
   * addFlatListTextStyle?: TextStyle; - стиль текста для не выбранных элементов в списке **FlatList** 
   * addFlatListTextStylePressed?: TextStyle; стиль текста для выбранного элемента в списке **FlatList** - **selectedItem**
   * addFlatListTextFont?: TypographyProp; - шрифт для текста для не выбранных элементов в списке **FlatList**
   * addFlatListTextFontPressed?: TypographyProp; шрифт для текста для не выбранного элемента в списке **FlatList** - **selectedItem**
   * selectedItemColor?: string; - цвет выбранного элемента в списке при нажатии на него - подсветка **TouchableHighLight**
   * maxVisibleListLength?: number: - Максимальное количество видимых элементов в списке(Если элементов больше, вью у FlatList будет скроллится) - **по дефолту 6** 