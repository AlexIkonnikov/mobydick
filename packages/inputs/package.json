{
  "name": "@npm/mobydick-inputs",
  "version": "0.0.0-alpha.12",
  "description": "Inputs on MobyDick core",
  "main": "index.ts",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Kirill Lebedev <lebedevki@lad24.ru>",
  "license": "MIT",
  "peerDependencies": {
    "react": ">=17.0.2",
    "react-native": ">=0.66.4"
  },
  "dependencies": {
    "@npm/mobydick-styles": "0.0.0-alpha.12",
    "@npm/mobydick-core": "0.0.0-alpha.12",
    "@npm/mobydick-typography": "0.0.0-alpha.12",
    "@react-native-community/hooks": "^2.8.1",
    "react-native-device-info": "9.0.2"
  },
  "publishConfig": {
    "@npm:registry": "https://glab.lad24.ru/api/v4/projects/609/packages/npm/"
  },
  "repository": {
    "type": "git",
    "url": "https://glab.lad24.ru/npm/mobydick.git",
    "directory": "packages/inputs"
  }
}