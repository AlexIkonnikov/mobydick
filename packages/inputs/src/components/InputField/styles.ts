import {StyleSheet} from 'react-native';
import {useTheme} from '@mobydick/styles';

const styles = (theme: ReturnType<typeof useTheme>) =>
  StyleSheet.create({
    textInput: {
      backgroundColor: theme.BgSecondary,
      padding: 10, // TODO: Брать из темы, когда они будут готовы
      borderRadius: 8, // TODO: Брать из темы, когда они будут готовы
      minWidth: 150, // Не уверен, что это нам надо
    },
    label: {
      marginBottom: 10, // TODO: Брать из темы, когда они будут готовы
    },
    underInputText: {
      color: theme.TextMuted,
      fontSize: 12, // TODO: Брать из темы, когда они будут готовы
    },
  });
export default styles;
