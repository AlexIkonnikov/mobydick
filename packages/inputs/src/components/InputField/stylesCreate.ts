import {IThemeContext, rem} from '@npm/mobydick-styles';
import {StyleSheet} from 'react-native';

import {IInputsTypes} from '../types';
import {disabledStyle, validStyle, wrongStyle} from '../../style';

const defaultStyle = (theme: IThemeContext, focused: boolean) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    container: {
      minWidth: rem(130),
    },
    inputContainer: {
      backgroundColor: colors.BgSecondary,
      paddingLeft: rem(18),
      paddingRight: spaces.Space16,
      borderRadius: spaces.Space8,
      borderWidth: spaces.Space1,
      borderColor: focused ? colors.BorderNormal : 'transparent',
      marginVertical: spaces.Space8,
      height: rem(48),
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      marginBottom: 50,
    },
    textInput: {
      flex: 1,
      padding: 0, // Android по дефолту ставит padding на input's
      fontSize: spaces.Space16,
      color: colors.TextPrimary,
      lineHeight: spaces.Space20,
    },
  });
};

const stylesCreate = (
  theme: IThemeContext,
  type: IInputsTypes,
  focused: boolean,
) => {
  switch (type) {
    case IInputsTypes.valid:
      return validStyle(theme, defaultStyle(theme, focused), focused);
    case IInputsTypes.wrong:
      return wrongStyle(theme, defaultStyle(theme, focused), focused);
    case IInputsTypes.disabled:
      return disabledStyle(theme, defaultStyle(theme, focused), focused);
    case IInputsTypes.default:
    default:
      return defaultStyle(theme, focused);
  }
};
export default stylesCreate;
