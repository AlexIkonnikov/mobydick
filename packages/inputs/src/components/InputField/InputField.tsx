import {ITextInput, TextInput, View} from '@npm/mobydick-core';
import React, {forwardRef, useMemo, useState} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {useFont} from '@npm/mobydick-typography';
import {Platform} from 'react-native';

import {IInputsTypes} from '../types';
import {InputSubtitle, InputTitle} from '../Base';

import {IInputFieldsProps} from './types';
import stylesCreate from './stylesCreate';
import Constants from './constants';

/**
 * Пример использования ref:
 * ```
 * const biba = () => {
 *   const ref = useRef<ITextInput>(null);
 *   const boba = () => {
 *     ref.current?.isFocused();
 *   };
 *   return (
 *     <InputField ref={ref} /> />
 *   );
 * };
 * ```
 */
const InputField = forwardRef<ITextInput, IInputFieldsProps>((props, ref) => {
  const {
    style,
    title,
    subtitle,
    rightIcon,
    type = IInputsTypes.default,
    disabled = false,
    titleProps,
    subtitleIcon,
    subtitleProps,
    containerStyle,
    textInputContainerStyle,
    onFocus,
    onBlur,
    required,
    secureTextEntry,
    ...otherProps
  } = props;
  const [focused, setFocused] = useState(false);
  const [styles, theme] = useStyles(
    stylesCreate,
    disabled ? IInputsTypes.disabled : type,
    focused,
  );
  const {fontStyle} = useFont('Regular-Primary-M');

  const getStyle = useMemo(() => {
    if (secureTextEntry && Platform.OS === 'android') {
      return [styles.secureTextInput];
    } else {
      return fontStyle;
    }
  }, [secureTextEntry, fontStyle.color]);

  return (
    <View style={[styles.container, containerStyle]}>
      {title && (
        <InputTitle title={title} titleProps={titleProps} required={required} />
      )}

      <View style={[styles.inputContainer, textInputContainerStyle]}>
        <TextInput
          ref={ref}
          testID={Constants.testID}
          style={[styles.textInput, getStyle, style]}
          placeholderTextColor={theme.colors.TextMuted}
          editable={!disabled}
          numberOfLines={1}
          onFocus={event => {
            setFocused(true);
            onFocus?.(event);
          }}
          onBlur={event => {
            setFocused(false);
            onBlur?.(event);
          }}
          selectionColor={theme.colors.IconBase}
          secureTextEntry={secureTextEntry}
          {...otherProps}
        />
        {rightIcon}
      </View>
      {subtitle ? (
        <InputSubtitle
          type={type}
          subtitle={subtitle}
          subtitleIcon={subtitleIcon}
          subtitleProps={subtitleProps}
        />
      ) : null}
    </View>
  );
});

export default InputField;
