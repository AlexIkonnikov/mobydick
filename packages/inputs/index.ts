import InputField from './src/components/InputField';

export * from './src/components/InputField/types';

export {InputField};
