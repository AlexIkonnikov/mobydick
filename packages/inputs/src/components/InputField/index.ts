import InputField from './InputField';

export * from './types';

export {InputField};
