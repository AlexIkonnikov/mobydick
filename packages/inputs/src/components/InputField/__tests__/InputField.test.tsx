import React from 'react';
import {act, render} from '@testing-library/react-native';
import {ITypes} from '@mobydick/inputs';

import InputField from '../index';
import Constants from '../constants';

describe('@mobydick/inputs/InputField', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <InputField title={'title'} subtitle={'subtitle'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly valid', () => {
    const {toJSON, getByTestId} = render(<InputField type={ITypes.valid} />);

    expect(toJSON()).toMatchSnapshot();

    const TextInput = getByTestId(Constants.testID);
    act(() => TextInput.props.onFocus());

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly wrong', () => {
    const {toJSON, getByTestId} = render(<InputField type={ITypes.wrong} />);

    expect(toJSON()).toMatchSnapshot();

    const TextInput = getByTestId(Constants.testID);
    act(() => TextInput.props.onFocus());

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly disabled', () => {
    const {toJSON, getByTestId} = render(
      <InputField type={ITypes.disabled} disabled={true} />,
    );

    expect(toJSON()).toMatchSnapshot();

    const TextInput = getByTestId(Constants.testID);
    act(() => TextInput.props.onFocus());

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly focused/blur', () => {
    const {toJSON, getByTestId} = render(<InputField />);
    const TextInput = getByTestId(Constants.testID);
    act(() => TextInput.props.onFocus());

    expect(toJSON()).toMatchSnapshot();

    act(() => TextInput.props.onBlur());

    expect(toJSON()).toMatchSnapshot();
  });
});
