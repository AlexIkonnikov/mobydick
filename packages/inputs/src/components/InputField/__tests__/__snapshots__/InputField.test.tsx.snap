// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-inputs/InputField renders correctly 1`] = `
<View>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        undefined,
      ]
    }
  >
    title
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Object {
          "color": "#9BA1B0",
          "fontFamily": "Neotis",
          "fontSize": 24,
          "marginRight": 6,
        }
      }
    >
      
    </Text>
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#9BA1B0",
            "fontFamily": "Inter-Regular",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly disabled 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#D1D4DB",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly disabled 2`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#D1D4DB",
          "borderColor": "#D1D4DB",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly focused/blur 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.3)",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly focused/blur 2`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly valid 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(43, 120, 238, 0.15)",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly valid 2`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(43, 120, 238, 0.15)",
          "borderColor": "#2B78EE",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly wrong 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(245, 77, 61, 0.15)",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Object {
          "color": "#F54D3D",
          "fontFamily": "Neotis",
          "fontSize": 24,
          "marginRight": 6,
        }
      }
    >
      
    </Text>
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#F54D3D",
            "fontFamily": "Inter-Regular",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/InputField renders correctly wrong 2`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(245, 77, 61, 0.15)",
          "borderColor": "#F54D3D",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "minWidth": 195,
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      style={
        Array [
          Object {
            "flex": 1,
            "padding": 0,
          },
          Object {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
          },
          undefined,
        ]
      }
      testID="TextInput"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Object {
          "color": "#F54D3D",
          "fontFamily": "Neotis",
          "fontSize": 24,
          "marginRight": 6,
        }
      }
    >
      
    </Text>
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#F54D3D",
            "fontFamily": "Inter-Regular",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;
